src/python/geoclaw/surge/storm.py

```diff
@@ -113,22 +116,197 @@ def read_geoclaw(self, path):
+                        lon = float(lon[0:-1]) 
+                    else: 
+                        lon = -1*float(lon[0:-1])
+                    self.eye_location[n,0] = lat 
+                    self.eye_location[n,1] = lon
+            data_file.close() 
+ 
     def read_imd(self, path):
-        r""""""
-        raise NotImplementedError("IMD format not fully implemented.")
+        r"""Extract relevant hurricane data from IMD file
+            and update storm fields with proper values. 
+        
+        :Input: 
+         - *path* (string) Path to the file to be read.
+
+        Return ValueError if format incorrect or if file not IMD.
+        """
+        if file_format.upper() not in _supported_formats:
+            raise ValueError("File type not one of supported formats.")
+        else:
+            raise ValueError("File type not implemented yet.") 
+ 
+    def read_tcvitals(self, path):
+        r"""Extract relevant hurricane data from TCVITALS file
+            and update storm fields with proper values. 
+        
+        :Input: 
+         - *path* (string) Path to the file to be read.
+
+        Return ValueError if format incorrect or if file not TCVITALS.
+        """
+        if file_format.upper() not in _supported_formats:
+            raise ValueError("File type not one of supported formats.")
+        else:
+            with open(path,'r') as data_file:
+                # Collect data from columns of the same type 
+                data = numpy.genfromtxt(path,dtype=None,usecols=(8,9,11,13))
+                self.max_wind_speed = data[:,0]                # Col 8 TCVITALS
+                self.central_pressure = data[:,1]              # Col 9 TCVITALS
+                self.storm_radius = data[:,2]                  # Col 11 TCVITALS
+                self.max_wind_radius = data[:,3]               # Col 13 TCVITALS
+
+                date = numpy.genfromtxt(path,dtype=None,usecols=(0,1))
+                self.new_year = int(str(date[0])[0:4])  
+                for i in range(date.shape[0]): 
+                    temp_date = "%s%s" %(date[i][0],date[i][1])
+                    temp_date = date2seconds(temp_date[0:-2])
+                    date[i][0] = temp_date 
+                self.t = date[:,0] 
+
+                self.eye_location = numpy.genfromtxt(path,dtype=None,usecols=(4,5))  
+                for n in range(self.eye_location.shape[0]): 
+                    lat = self.eye_location[n,0]
+                    lon = self.eye_location[n,1]
+                    if lat[-1] == 'N':
+                        lat = float(lat[0:-1])/10 
+                    else: 
+                        lat = -1*float(lat[0:-1])/10
+                    if lon == 'E': 
+                        lon = float(lon[0:-1])/10
+                    else: 
+                        lon = -1*float(lon[0:-1])/10
+                    self.eye_location[n,0] = lat 
+                    self.eye_location[n,1] = lon
+            data_file.close() 
```

Comment:
You do not need to explicitly close `data_file` as it does so implicitly at the end of a `while` block.